### PR TITLE
Add zh_CN translations

### PR DESCRIPTION
**Pull request by ouuan**
#428

### CODE REVIEW

**Comment by Morganamilo** [comment]
Thanks for being the first. Fixed up the stuff you mentioned so you need to update the translations as shown in contribution (there's no need to update the pot file though so just `./scripts/updpo`. It's good that it gets tested before being merged.

Also have you tested `paru -Si` with this? My assumption is all the alignment is thrown off given the characters are multi byte. It's something I am going to tackle after #428 is merged.

**Comment by ouuan** [comment]
> Also have you tested `paru -Si` with this? My assumption is all the alignment is thrown off given the characters are multi byte. It's something I am going to tackle after #428 is merged.

图片

**Comment by Morganamilo** [comment]
Yep as expected. Now I have output to test against :p

**Comment by Morganamilo** [comment]
Also do you think AUR and URL should be localised?

**Comment by ouuan** [comment]
> Also do you think AUR and URL should be localised?

It's not localized in yay :thinking: I don't think "AUR" should be localized. "URL" doesn't matter.

**Comment by Morganamilo** [comment]
Okay. Looks a bit funny with the mix of Chinese and not Chinese. But I guess that's just how it is.

File: po/zh_CN.po [hunk 1]
```diff
@@ -0,0 +1,706 @@
...
+msgid "only one operation may be used at a time"
+msgstr "一次只能使用一个操作"
+
+#: src/config.rs:638
+msgid "can not find local repo '{}' in pacman.conf"
+msgstr "无法找到 pacman.conf 中的本地仓库 '{}'"
+
+#: src/config.rs:670
+msgid "failed to initialize alpm: root={} dbpath={}"
+msgstr "未能成功初始化 alpm: root={} dbpath={}"
```
Comment (lilydjwg) [inline po/zh_CN.po:106]: 「成功」不需要。

File: po/zh_CN.po [hunk 2]
```diff
@@ -0,0 +1,706 @@
...
+
+#: src/config.rs:740 src/config.rs:873
+msgid "value can not be empty for key '{}'"
+msgstr "选项 '{}' 的值不能为空"
+
+#: src/config.rs:756
+msgid "key '{}' does not belong to a section"
+msgstr "选项 '{}' 不属于任何一个设置板块"
```
Comment (lilydjwg) [inline po/zh_CN.po:114]: key: 键； section: 节。

File: po/zh_CN.po [hunk 3]
```diff
@@ -0,0 +1,706 @@
...
+msgid "does not contain PKGBUILD: skipping"
+msgstr "不包含 PKGBUILD: 正在跳过"
+
+#: src/download.rs:294
+msgid " ({:total>padding$}/{:total}) downloading: {:pkg}"
+msgstr " ({:total>padding$}/{:total}) 正在下载 {:pkg}"
+
+#: src/download.rs:317
+msgid "Downloading PKGBUILDs..."
+msgstr "正在下载 PKGBUILDs..."
```
Comment (lilydjwg) [inline po/zh_CN.po:219]: s 去掉。

File: po/zh_CN.po [hunk 4]
```diff
@@ -0,0 +1,706 @@
...
+msgid "no architecture"
+msgstr "无架构"
+
+#: src/install.rs:138
+msgid "can't build package as root"
+msgstr "无法以 root 身份构建软件包"
+
+#: src/install.rs:142
+msgid "failed to parse srcinfo generated by makepkg"
+msgstr "未能成功解析由 makepkg 生成的 srcinfo"
```
Comment (lilydjwg) [inline po/zh_CN.po:343]: 「成功」不需要。

File: po/zh_CN.po [hunk 5]
```diff
@@ -0,0 +1,706 @@
...
+
+#: src/install.rs:171 src/install.rs:442
+msgid "Resolving dependencies..."
+msgstr "正在解析依赖关系..."
```
Comment (lilydjwg) [inline po/zh_CN.po:347]: resolve = 解决。

File: po/zh_CN.po [hunk 6]
```diff
@@ -0,0 +1,706 @@
...
+
+#: src/install.rs:220 src/install.rs:225
+msgid "failed to download sources"
+msgstr "未能成功下载源码"
+
+#: src/install.rs:230 src/install.rs:246 src/install.rs:254
+msgid "failed to build"
+msgstr "未能成功构建"
```
Comment (lilydjwg) [inline po/zh_CN.po:355]: 哪来这么多「成功」啊……

File: po/zh_CN.po [hunk 7]
```diff
@@ -0,0 +1,706 @@
...
+msgid "can't install AUR package as root"
+msgstr "无法以 root 身份安装 AUR 包"
+
+#: src/install.rs:507
+msgid "Remove make dependencies after install?"
+msgstr "在安装后移除生成依赖？"
+
+#: src/install.rs:518
+msgid "Proceed to review?"
+msgstr "继续以审核？"
```
Comment (lilydjwg) [inline po/zh_CN.po:396]: 「审阅」。

File: po/zh_CN.po [hunk 8]
```diff
@@ -0,0 +1,706 @@
...
+#: src/install.rs:559
+msgid "The following packages are not compatible with your architecture:"
+msgstr "下列软件包与你的系统架构不兼容:"
+
+#: src/install.rs:573
+msgid "Would you like to try build them anyway?"
+msgstr "你还是想要构建它们吗？"
+
+#: src/install.rs:662
+msgid "packages failed to buid: {}"
```
Comment (lilydjwg) [inline po/zh_CN.po:407]: Here's a typo.

**Comment by ouuan** [comment]
Is there any difference between "Looking for devel repos..." and "fetching devel info..."?

File: po/zh_CN.po [hunk 9]
```diff
@@ -0,0 +1,706 @@
...
+
+#: src/install.rs:1768
+msgid ""
+"\n"
+"\n"
+"Enter a selection (default=all): "
+msgstr ""
+"\n"
+"\n"
+"输入某个选择 (默认=全部选定): "
```
Comment (lilydjwg) [inline po/zh_CN.po:546]:
```suggestion
"输入选择 (默认=全部选定): "
```
「某个」听起来只能选一个。

**Comment by ouuan** [comment]
I think it's difficult to translate "devel". Can I translate it to something like "upstream repository"?

File: po/zh_CN.po [hunk 10]
```diff
@@ -0,0 +1,706 @@
...
+msgid "Aur packages: {}"
+msgstr "AUR 包: {}"
+
+#: src/stats.rs:91
+msgid "Repo packages: {}"
+msgstr "软件库包: {}"
+
+#: src/stats.rs:95
+msgid "Explicitly installed packages: {}"
+msgstr "单独指定安装的包: {}"
```
Comment (lilydjwg) [inline po/zh_CN.po:630]:
```suggestion
msgstr "显式指定安装的包: {}"
```

File: po/zh_CN.po [hunk 11]
```diff
@@ -0,0 +1,706 @@
...
+"Content-Transfer-Encoding: 8bit\n"
+"X-Generator: Poedit 3.0\n"
+
+#: src/clean.rs:31
+msgid "Do you want to remove ALL AUR packages from cache?"
+msgstr "你想要从缓存中移除所有 AUR 包吗？"
+
+#: src/clean.rs:33
+msgid "Do you want to remove all other AUR packages from cache?"
+msgstr "你想要从缓存中移除所有未安装的 AUR 包吗？"
```
[the remark this answers is not in the document]
Comment (ouuan) [reply po/zh_CN.po:26]: > 「未安装」是哪里来的呢？

你是说原文没有？好像实际行为如此。

File: po/zh_CN.po (hunk 9 above)
Comment (ouuan) [reply po/zh_CN.po:546]: > 「某个」听起来只能选一个。

我是从 pacman 复制来的 :thinking:

File: po/zh_CN.po (hunk 5 above)
Comment (ouuan) [reply po/zh_CN.po:347]: 是从 pacman 复制过来的

File: po/zh_CN.po (hunk 10 above)
Comment (ouuan) [reply po/zh_CN.po:630]: 这里也是从 pacman 复制过来的

**Comment by Morganamilo** [comment]
> Is there any difference between "Looking for devel repos..." and "fetching devel info..."?

Same thing really. Just one is during install and once is during --gendb

One would have to be capitalised anyway due to convention so I don't see a point setting them to the same string in code.

**Comment by Morganamilo** [comment]
> I think it's difficult to translate "devel". Can I translate it to something like "upstream repository"?

In this context devel is short for development. Packages that always get the latest version instead of a fixed release. Usually -git, sometimes other stuff like -always, -latest, -hg.

Though I guess you are technically right because what paru does when "checking devel updates" and such is ping the upstream repo. But I see that as an implementation detail and not really what it should say.

**Comment by ouuan** [comment]
> In this context devel is short for development. Packages that always get the latest version instead of a fixed release. Usually -git, sometimes other stuff like -always, -latest, -hg.
> 
> Though I guess you are technically right because what paru does when "checking devel updates" and such is ping the upstream repo. But I see that as an implementation detail and not really what it should say.

So the point is that the version is the development version, not that paru is checking the upstream repository? I thought "devel" was related to VCS (something like "development repository"). If it means the development version, I can translate it in this way.

**Comment by ouuan** [comment]
> Same thing really. Just one is during install and once is during --gendb
> 
> One would have to be capitalised anyway due to convention so I don't see a point setting them to the same string in code.

I wanted to know how to translate them, so it's confusing when I see two different sentences for the same thing.

**Comment by Morganamilo** [comment]
> So the point is that the version is the development version, not that paru is checking the upstream repository? I thought "devel" was related to VCS (something like "development repository"). If it means the development version, I can translate it in this way.

devel packages are usually in some sort of VCS but can sometimes not be for example firefox-nightly. "devel packages" is not really an official term, it's just what most aur helpers have diverged on.

> I wanted to know how to translate them, so it's confusing when I see two different sentences for the same thing.

Well I guess they are a bit different. `--gendb` scans all of your installed packages for upstream git repos` so "Looking for devel repos..." works. During `-S` it only looks for devel info in the packages that are to be installed so "fetching devel info...".

I guess you can think of it as a difference of scope. The former is "search the entire system for devel packages" and the later is "fetch the devel info for these packages".

Though that's a bit pretentious with the difference.

File: po/zh_CN.po (hunk 11 above)
[the remark this answers is not in the document]
Comment (ouuan) [reply po/zh_CN.po:26]: 看了下 pacman 也是“您想从缓存中删除所有其他的软件包吗？”，那我改一下吧

File: po/zh_CN.po (hunk 9 above)
Comment (ouuan) [reply po/zh_CN.po:546]: 「某个」确实听起来有点像是只能选一个，而且这里不是术语之类的，和 pacman 不同也没关系。

**Comment by Morganamilo** [comment]
Everything is in place. paru-git has been updated and -Si fixed. Thanks for setting the example.

**Pull request merged**